docs/REST API/rest-api.md

```diff
@@ -7711,11 +7821,163 @@ Deletes the session specified by ID. When the user associated with the session n
 
 ## Software
 
+- [Add software](#add-software)
+- [Download software](#download-software)
+- [Delete software](#delete-software)
+- [Get installation result](#get-installation-result)
 - [List software](#list-software)
 - [List software versions](#list-software-versions)
 - [Get software](#get-software)
 - [Get software version](#get-software-version)
 
+### Add software
+
+Add software package to install on macOS, Windows, and Linux (Ubuntu) hosts.
+
+`POST /api/v1/fleet/software/package`
```

Comment:
```suggestion
`POST /api/v1/fleet/software`
```

Maybe we can simplify this?

Do we need to specify `/package` here? Are you thinking this sets ourselves up for future endpoints? Which ones?

Up to you.

Reply:
The reason why I added `/package` is because of the `/download` endpoint. 

Currently download endpoint is:

GET /api/v1/fleet/software/package/:id?alt=media

If we in the future decide to have endpoint to get package info/metadata (`GET /api/v1/fleet/software/package/:id`), and remove `/package` we'll end up with `GET /api/v1/fleet/software` which is used for versions before.

Another reason is because the delete endpoint. We just delete a package not software, so users might be confused. I think we should be explicit `/delete/package`